### PR TITLE
Add documentation for checks

### PR DESCRIPTION
**Pull request by slesaad**
**Motivation**
To make pyQuARC easier for the users, it is set up in a way that a new rule can be added just by adding entries to a set of JSON files. A new rule basically points to a checks that is already implemented. 

We need documentation of all the available checks so the user can just go through it and find the check that they need to create the new rule. It should include the description of what the check does and the check id that can be used in the JSON files.

**Changes made**
Restructured the schema files `rule_mapping.json` and `checks.json`, such that the `checks` can be reused and users have to only make changes in the `rule_mapping.json` file.
Changes include:
- Moved `data` from `checks.json` to `rule_mapping.json`
- Added `check_id` to `rule_mapping.json` to point to already existing check (helps remove duplicates from `checks.json`)
- Removed the duplicate checks from `checks.json`

**Deliverable**
The documentation for all the available checks is provided in the file [CHECKS.md](https://github.com/NASA-IMPACT/pyQuARC/blob/feature-documentation/CHECKS.md)

### CODE REVIEW

**Comment by slesaad** [comment]
[pyQuARC: Documentation of the available checks (2)](https://trello.com/c/9YNqOP5b/1226-pyquarc-documentation-of-the-available-checks-2)

**Comment by code-geek** [comment]
LGTM

File: pyQuARC/code/checker.py [hunk 1]
```diff
@@ -158,10 +158,10 @@ def _run_func(self, func, rule, rule_id, metadata_content, result_dict):
         Run the check function for `rule_id` and update `result_dict`
         """
         dependencies = rule.get("dependencies", [])
-        external_data = rule.get("data", [])
         rule_mapping = self.rules_override.get(
             rule_id
         ) or self.rule_mapping.get(rule_id)
+        external_data = rule_mapping.get("data", [])
```
Comment (xhagrg) [inline pyQuARC/code/checker.py:164]: That was my bad 😬

**Review by xhagrg — APPROVED** [review]
Lgtm

**Pull request merged**